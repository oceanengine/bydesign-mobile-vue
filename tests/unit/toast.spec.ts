import { later } from '../index';
import { Toast as NativeToast } from '@src';
const Toast = NativeToast as any;

describe('Toast', () => {
  test('create a forbidClick toast', async () => {
    const toast = Toast({
      forbidClick: true,
      type: 'success',
    });

    await later();
    expect(toast.$el.outerHTML).toMatchSnapshot();

    await later();
    expect(document.body.classList.contains('byted-toast--unclickable')).toBeTruthy();
    toast.forbidClick = false;

    await later();
    expect(document.body.classList.contains('byted-toast--unclickable')).toBeFalsy();
  });
  test('toast disappeared after duration', async () => {
    const toast = Toast({
      duration: 500,
    });

    await later(1000);
    expect(toast.$el.style.display).toEqual('none');
  });

  test('show loading toast', async () => {
    const toast = Toast({
      message: 'loading...',
      type: 'loading',
    });

    await later();
    expect(toast.$el.outerHTML).toMatchSnapshot();
  });

  test('should trigger onClose callback after closed', () => {
    Toast.allowMultiple();
    const onClose = jest.fn();
    const toast = Toast({
      message: 'toast',
      onClose,
    });
    toast.clear();
    expect(onClose).toHaveBeenCalledTimes(1);
    Toast.allowMultiple(false);
  });

  test('trigger onOpened callback after open', async () => {
    const onOpened = jest.fn();
    const onClose = jest.fn();
    const toast = Toast({
      message: 'toast',
      duration: 100,
      onOpened,
      onClose,
    });
    await later(500);

    expect(onOpened).toHaveBeenCalledTimes(1);
    expect(onClose).toHaveBeenCalledTimes(1);
    toast;
  });
});
