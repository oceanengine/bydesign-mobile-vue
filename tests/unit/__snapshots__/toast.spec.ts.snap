// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast create a forbidClick toast 1`] = `<div class="byted-toast byted-toast--middle byted-fade-enter byted-fade-enter-active" style="z-index: 2000;"><i class="byted-toast__icon" style="fill: #ffffff; width: 1.5rem; height: 1.5rem;">test-file-stub</i></div>`;

exports[`Toast show loading toast 1`] = `
<div class="byted-toast byted-toast--middle byted-fade-enter byted-fade-enter-active" style="z-index: 2002;"><i class="byted-toast__icon byted-loading-circle" style="fill: #0278FF; width: 1.5rem; height: 1.5rem;">test-file-stub</i>
  <div class="byted-toast__text">loading...</div>
</div>
`;
